Audit Fixes QS-S-2

parv3213 opened the pull request:
1. We have implemented the recommended change.
2. We do agree that the implementation for these functions are quite similar, but further abstraction would result into gas cost increase for the transactions involving these functions. 
3. Merging computeRewards() and _updateFarmRewardData() is very complicated and might add-up unnecessary gas cost post abstraction. 
4. We have made deployerList and farms internal. We have kept the functions for making it easier to fetch data on frontend if required.
5. CamelotV3 and UniswapV3 farms involve different interfaces, we can try abstracting them, however it might result in unwanted complexities. Due to time constraints we will skip it for time being. 

<img width="1051" alt="Screenshot 2024-06-24 at 6 51 03 PM" src="https://github.com/Sperax/Demeter-Protocol/assets/45873074/c2427640-45ea-43c7-8eaa-403e1d38cc3f">

parv3213 committed:
perf: use existing `PREC` in place of `PRECISION`

arcantheon commented on contracts/features/OperableDeposit.sol:
```diff
@@ -31,8 +31,6 @@ import {Subscription, RewardFund, Deposit} from "../interfaces/DataTypes.sol";
 /// @author Sperax Foundation.
 /// @notice This contract helps in creating farms with increase/decrease deposit functionality.
 abstract contract OperableDeposit is ExpirableFarm {
-    uint256 public constant PRECISION = 1e18;
```
Maybe update the name to PRECISION in FarmStorage because PREC is a little confusing to outsiders and not so clear.

parv3213 replied:
Fixed in [6b82c96](https://github.com/Sperax/Demeter-Protocol/pull/78/commits/6b82c96167c1d139bec9f6d727eccad7628fd57d)

arcantheon reviewed (changes requested):
Review comment posted

parv3213 committed:
perf: rename `PREC` to `PRECISION` for clarity

parv3213 committed:
perf: caching lpToken, token0 and token1 for further gas-optimization

parv3213 committed:
Revert "perf: caching lpToken, token0 and token1 for further gas-optimization"

This reverts commit ff5646acd8b195a9768f2ab7f00a778999058fa5.

YashP16 commented:
@parv3213, Lets address all the other points here itself.

parv3213 committed:
fix: rebase missed changes

parv3213 committed:
feat: Make `farms` and `deployerList` internal

The code changes add two internal arrays, `farms` and `deployerList`, to the `FarmRegistry` contract. These arrays will be used to track the registered farms and deployers respectively. This change is necessary to improve the efficiency and organization of the contract.

parv3213 committed:
Merge branch 'dev' into fix/remove-redundencies
